com.unity.render-pipelines.high-definition/Runtime/Water/WaterLighting.compute

```diff
@@ -0,0 +1,134 @@
+RW_TEXTURE2D_X(float4, _CameraColorTextureRW);
+RW_TEXTURE2D_X(float4, _NormalBufferRW);
+
+[numthreads(8, 8, 1)]
+void WaterPrepareSSR(uint3 dispatchThreadId : SV_DispatchThreadID, uint2 groupThreadId : SV_GroupThreadID, uint2 groupId : SV_GroupID)
+{
+    UNITY_XR_ASSIGN_VIEW_INDEX(dispatchThreadId.z);
+
+    // Compute the pixel position to process
+    uint2 currentCoord = groupId * 8 + groupThreadId;
+
+    // Is this a water pixel?
+    uint stencilValue = GetStencilValue(LOAD_TEXTURE2D_X(_StencilTexture, currentCoord));
+    if ((stencilValue & STENCILUSAGE_WATER_SURFACE) == 0)
+        return;
+
+    // Read the depth value
+    float depthValue = LOAD_TEXTURE2D_X(_DepthTexture, currentCoord).x;
+
+    // If this is a background pixel or an invalid ray, leave right away
+    if (depthValue == UNITY_RAW_FAR_CLIP_VALUE)
+        return;
+
+    // Decode the normal and the smoothness
+    NormalData normalData;
+    DecodeWaterFromNormalBuffer(currentCoord, normalData);
+
+    // TODO: change when supporting non horizontal water surfaces
+    normalData.normalWS.y = abs(normalData.normalWS.y);
+
+    // Encode the normal data and output it
+    float4 outputNormalData;
+    EncodeIntoNormalBuffer(normalData, outputNormalData);
+    _NormalBufferRW[COORD_TEXTURE2D_X(currentCoord)] = outputNormalData;
+}
+
+[numthreads(8, 8, 1)]
+void WaterDeferredLighting(uint3 dispatchThreadId : SV_DispatchThreadID, uint2 groupThreadId : SV_GroupThreadID, uint2 groupId : SV_GroupID)
+{
+    UNITY_XR_ASSIGN_VIEW_INDEX(dispatchThreadId.z);
+
+    // Compute the pixel position to process
+    uint2 currentCoord = groupId * 8 + groupThreadId;
+
+    // Is this a water pixel?
+    uint stencilValue = GetStencilValue(LOAD_TEXTURE2D_X(_StencilTexture, currentCoord));
+    if ((stencilValue & 16) == 0)
+        return;
+
+    // Read the depth value
+    float depthValue = LOAD_TEXTURE2D_X(_DepthTexture, currentCoord).x;
+
+    // If this is a background pixel or an invalid ray, leave right away
+    if (depthValue == UNITY_RAW_FAR_CLIP_VALUE)
+        return;
+
+    // Compute the posInput
+    PositionInputs posInput = GetPositionInput(currentCoord, _ScreenSize.zw, depthValue, UNITY_MATRIX_I_VP, GetWorldToViewMatrix(), 0);
+
+    // Compute the view vector
+    float3 V = GetWorldSpaceNormalizeViewDir(posInput.positionWS);
+
+    // Read the bsdf data and builtin data from the gbuffer
+    BSDFData bsdfData;
+    ZERO_INITIALIZE(BSDFData, bsdfData);
+
+    BuiltinData builtinData;
+    ZERO_INITIALIZE(BuiltinData, builtinData);
+
+    // Decode the water data from the gbuffer
+    DecodeFromGBuffer(currentCoord, bsdfData, builtinData);
+
+    // Compute the prelight data
+    PreLightData preLightData = GetPreLightData(V, posInput, bsdfData);
+
+    // Evaluate the light loop
+    uint  featureFlags = UINT_MAX;
+    LightLoopOutput lightLoopOutput;
+    LightLoop(V, posInput, preLightData, bsdfData, builtinData, featureFlags, lightLoopOutput);
+
+    // Alias
```

Comment:
note for myself: what differ from regular deffered.compute?

Reply:
Decode function (due to the different in light layer mangement), lighting model includes, fog, material variants, no usage of the material tile classification or light classification plus things that could be refactored (stencil test, etc.)

It would be a bit of a workout to make it work just to share the file